WorldOfPAYBACK/Services/TransactionsDataService.swift

```diff
@@ -12,24 +12,63 @@ class TransactionsDataService: ObservableObject {
 
     @Published var transactions: [Item] = []
     @Published var error: StaticJSONMapper.MappingError?
+    @Published var errorAPI: APIManager.NetworkingError?
     @Published var hasError: Bool = false
-
+    @Published private(set) var isLoading = true
     var jsonFileSubscription: AnyCancellable?
 
     init() {
         fetchFromFile()
+//        fetchFromPROD()
+//        fetchFromTEST()
+    }
+
+    func fetchFromFile() {
+        do {
+            let jsonFileSubscription = try StaticJSONMapper.decode(file: "PBTransactions", type: Transactions.self)
+            transactions = jsonFileSubscription.items
+            isLoading = false
+        } catch {
+            print(error)
+            self.hasError = true
+            self.error = error as? StaticJSONMapper.MappingError
+        }
     }
 
-            func fetchFromFile() {
-                do {
-                    let jsonFileSubscription = try StaticJSONMapper.decode(file: "PBTransactions", type: Transactions.self)
-                    transactions = jsonFileSubscription.items
-                } catch {
-                    print(error)
-                    self.hasError = true
-                    self.error = error as? StaticJSONMapper.MappingError
-                }
+    func fetchFromPROD() async {
```

Comment:
Fetch form at a minimum should be a parameter. Now you have two almost identical methods.